### DISCUSSION TITLE
Obok plugin not seeing recent books

### DISCUSSION THREAD

**Discussion by dabneykr** [noDRM/DeDRM_tools#625] [Q&A] [answered]
Tried to import my most recent books purchased in the last few days and they are not showing up in the Obok import list. True for both the 10.0.9 version and the 10.0.3 version. Running on Calibre version 7.4.0. A reinstall of the plugin didn't change anything, and I double checked the books were downloaded in the Kobo app. One is a DRM free EPUB 3 and the other is a EPUB 3 but with Adobe DRM placed on it. This portion does not have an error code.

I tried manually importing the DRM free book, but the way it is stored in Kobo wouldn't let it be seen as a book. 

Failed to add the file C:\Users\Kyoti\AppData\Local\Kobo\Kobo Desktop Edition\kepub\0c3f9547-75f7-4378-9aa3-15e6a8dbeff3 to the book: 0c3f9547-75f7-4378-9aa3-15e6a8dbeff3
With error:
Traceback (most recent call last):
  File "calibre\gui2\add.py", line 491, in add_formats
  File "calibre\db\cache.py", line 1981, in add_format
  File "calibre\db\cache.py", line 1937, in _do_add_format
  File "calibre\db\backend.py", line 1836, in add_format
  File "calibre\db\backend.py", line 1484, in construct_file_name
ValueError: Extension length too long: 98

I am at a loss of how to approach this to easily import the files.

**Comment by ElleKayEm** [comment, marked as answer]
Try closing Kobo Desktop before using Obok.  Alternatively you could download from the Kobo website.  The DRM-free book will be a regular epub.  For the Adobe DRM book, you will get an ACSM file.  Open that with Adobe Digital Editions authorized with an Adobe ID to get the epub.  Then DeDRM can remove that DRM.

To manually import the DRM-free Kobo Desktop book, you'd need to rename it and give an epub extension.

**Reply by dabneykr** [reply]
Thank you so much! Apparently the easy answer of 'close Kobo first' made a huge difference. Thank you!

**Reply by ElleKayEm** [reply]
Glad it was an easy fix!
